### DISCUSSION TITLE
"port already in user" error when sing with npm in same stack

### DISCUSSION THREAD

**Discussion by vdrover** [alexjustesen/speedtest-tracker#554] [Q&A]
I am using ports "80:80" with npm in my stack. When I try to add speedtest-tracker, I get the error above in my docker logs and the "docker compose up" command halts. 

Any recommendations on using speedtest-tracker with npm when npm is using port 80 already so that docker compose will continue as expected?

**Comment by spcqike** [comment]
Just use another port. Like 82:80. 

I’m using 8008:80, NPM than works as proxy and serves „HTTP://IP:8008“ as „speedtracker.local“ when I try to access it. 

Of course you can’t have two services running on the same port. So you either need a different host or a different port. 

i guess with NPM you mean nginx proxy manager?  If so, you already have a working proxy to do everything. When done right, you don’t need to open a port on your host at all. (For your services. NPM still needs port 80 and 443, 81 too) NPM and your services just need to sit in the same docker network.
